### PR TITLE
Update config parsing

### PR DESCRIPTION
**Pull request by dscott90**

### CODE REVIEW

File: chipsec/hal/pci.py [hunk 1]
```diff
@@ -293,15 +293,19 @@ def enumerate_devices(self, bus: Optional[int] = None, device: Optional[int] = N
...
+                    did_vid = self.read_dword(b, d, f, 0x0)
+                    if 0xFFFFFFFF != did_vid:
+                        vid = did_vid & 0xFFFF
+                        did = (did_vid >> 16) & 0xFFFF
+                        rid = self.read_byte(b, d, f, 0x8)
+                        devices.append((b, d, f, vid, did, rid))
+                    elif f == 0 and spec:
+                        break
+                except OsHelperError:
+                    self.logger.log_hal("[pci] unable to access B/D/F: {:d}/{:d}/{:d}".format(b, d, f))
```
Comment (BrentHoltsclaw) [inline chipsec/hal/pci.py:308]: convert back to fstring

File: chipsec_main.py [hunk 2]
```diff
@@ -378,11 +378,11 @@ def properties(self):
         ret["OS"] = "{} {} {} {}".format(self._cs.helper.os_system, self._cs.helper.os_release,
                                          self._cs.helper.os_version, self._cs.helper.os_machine)
         ret["Python"] = "Python {}".format(platform.python_version())
-        ret["Platform"] = "{}, CPUID: {}, VID: {:04X}, DID: {:04X}, RID: {:02X}".format(self._cs.longname, self._cs.cpuid, self._cs.vid,
-                                                                                        self._cs.did, self._cs.rid)
+        ret["Platform"] = "{}, CPUID: {}, VID: {:04X}, DID: {:04X}, RID: {:02X}".format(self._cs.Cfg.longname, self._cs.Cfg.cpuid, self._cs.Cfg.vid,
```
Comment (BrentHoltsclaw) [inline chipsec_main.py:381]: fstring?

File: chipsec_main.py [hunk 3]
```diff
@@ -378,11 +378,11 @@ def properties(self):
...
+                                                                                        self._cs.Cfg.did, self._cs.Cfg.rid)
         if not self._cs.is_atom():
-            ret["PCH"] = "{}, VID: {:04X}, DID: {:04X} RID: {:02X}".format(self._cs.pch_longname, self._cs.pch_vid,
-                                                                           self._cs.pch_did, self._cs.pch_rid)
+            ret["PCH"] = "{}, VID: {:04X}, DID: {:04X} RID: {:02X}".format(self._cs.Cfg.pch_longname, self._cs.Cfg.pch_vid,
```
Comment (BrentHoltsclaw) [inline chipsec_main.py:384]: fstring?

File: chipsec/hal/pci.py [hunk 4]
```diff
@@ -277,7 +277,7 @@ def write_dword(self, bus: int, device: int, function: int, address: int, dword_
     # Enumerating PCI devices and dumping configuration space
     #
 
-    def enumerate_devices(self, bus: Optional[int] = None, device: Optional[int] = None, function: Optional[int] = None) -> List[Tuple[int, int, int, int, int]]:
+    def enumerate_devices(self, bus: Optional[int] = None, device: Optional[int] = None, function: Optional[int] = None, spec: bool = True) -> List[Tuple[int, int, int, int, int, int]]:
```
Comment (BrentHoltsclaw) [inline chipsec/hal/pci.py:280]: spec: Optional[bool] = True ?

File: chipsec/chipset.py [hunk 5]
```diff
@@ -155,176 +91,84 @@ def set_hal_objects(self):
...
-        self.pch_rid = 0xFF
-        
-        
+        # platform detection
+
+        # get cpuid only if driver using driver (otherwise it will cause problems)
+        #if start_driver or self.helper.is_linux():
+        self.cpuid = None
         if start_helper:
+            #cpuid = self.get_cpuid()
```
Comment (BrentHoltsclaw) [inline chipsec/chipset.py:128]: delete

File: chipsec/chipset.py [hunk 6]
```diff
@@ -155,176 +91,84 @@ def set_hal_objects(self):
...
             self.load_helper(helper_name)
             self.start_helper()
-            self.vid, self.did, self.rid, self.pch_vid, self.pch_did, self.pch_rid = self.read_platform_ids()
+            #self.vid, self.did, self.rid, self.pch_vid, self.pch_did, self.pch_rid = self.read_platform_ids()
```
Comment (BrentHoltsclaw) [inline chipsec/chipset.py:131]: delete

File: chipsec/chipset.py [hunk 7]
```diff
@@ -155,176 +91,84 @@ def set_hal_objects(self):
...
+
             # get cpuid only if using driver (otherwise it will cause problems)
-            self.get_cpuid()
 
-        
+            self.cpuid = self.get_cpuid() 
+        #else:
```
Comment (BrentHoltsclaw) [inline chipsec/chipset.py:136]: delete

File: chipsec_main.py (hunk 2 above)
Comment (npmitche) [reply chipsec_main.py:381]: We haven't swapped chipsec_main yet, so we'll do that in it's on PR at a later date.

File: chipsec_main.py (hunk 3 above)
Comment (npmitche) [reply chipsec_main.py:384]: Will resolve later.

**Review by npmitche — APPROVED** [review]

File: chipsec/hal/pci.py [hunk 8]
```diff
@@ -293,15 +293,19 @@ def enumerate_devices(self, bus: Optional[int] = None, device: Optional[int] = N
...
+                    did_vid = self.read_dword(b, d, f, 0x0)
+                    if 0xFFFFFFFF != did_vid:
+                        vid = did_vid & 0xFFFF
+                        did = (did_vid >> 16) & 0xFFFF
+                        rid = self.read_byte(b, d, f, 0x8)
+                        devices.append((b, d, f, vid, did, rid))
+                    elif f == 0 and spec:
+                        break
+                except OsHelperError:
+                    self.logger.log_hal(f"[pci] unable to access B/D/F: {b:d}/{d:d}/{f:d}")
```
Comment (npmitche) [inline chipsec/hal/pci.py:308]: Should switch this to print hex values.

**Review by npmitche — APPROVED** [review]
Good Job Thumbs Up
LGTM

**Review by BrentHoltsclaw — APPROVED** [review]

File: chipsec/cfg/8086/glk.xml [hunk 9]
```diff
@@ -9,7 +9,7 @@
   <!-- Information                          -->
   <!--                                      -->
   <!-- #################################### -->
-  <info family="atom" detection_value="706ax">
```
Comment (frinzell) [inline chipsec/cfg/8086/glk.xml:12]: Recommend making this a ticket.

**Comment by npmitche** [comment]
Rebased and resolved conflicts.

File: chipsec/cfg/8086/glk.xml [hunk 10]
```diff
@@ -9,7 +9,7 @@
   <!-- Information                          -->
   <!--                                      -->
   <!-- #################################### -->
-  <info family="atom" detection_value="706ax">
+  <info family="atom" detection_value="0x7006a"> <!-- verify detection val -->
```
Comment (fishilico) [inline chipsec/cfg/8086/glk.xml:12]: Hello @dscott90 , this change seems to have reverted what https://github.com/chipsec/chipsec/pull/1735 ("Fix CPUID detection value of Gemini Lake (GLK) ") did: it replaced `7006a` with `706ax`, providing example of values.

Was there something wrong with using `0x706a0-0x706af` as the new `detection_value` for GLK?

Comment (npmitche) [reply chipsec/cfg/8086/glk.xml:12]: There was a conflict with what we were merging in and we made a note to verify. Looks like we chose the wrong one. Thanks for pointing it out. I just opened a PR #1910  to re-fix it.

Comment (fishilico) [reply chipsec/cfg/8086/glk.xml:12]: Thanks for your quick reply! The PR looks good to me.

**Pull request merged**